### ISSUE TITLE
Delivery details unauthorized for receivers

### ISSUE THREAD

**Issue by kevinhaube** [CDCgov/prime-reportstream#7578]
## Problem statement

As a receiver on the ReportStream React app, I receive a 401 Unauthorized error when accessing Delivery Details from the Daily Data list page.
![Screen Shot 2022-12-05 at 4.34.21 PM.png](https://images.zenhubusercontent.com/611a907ec3e013d0e33e58a7/256753d9-2e84-4a2c-9955-03b7ff889b36)

## Recreate

1. Login with _NO_ PrimeAdmin group, only a receiver group, on your Okta account
2. View Daily Data
3. Click through one of the IDs to the details page
4. You will get 3-4 401's as it retries the call on each failure until it hits its limit

## Expected behavior

- [ ] Endpoint validates the organization to see its own reports

## To do

- [ ] ...

**Comment by kevinhaube** [comment]
Happy to coordinate with whomever takes this to set up their Okta accounts to reflect this bugs replication requirements.

**Comment by pawansapra01** [comment]
@kevinhaube could you help with the Okta accounts setup for this?

**Comment by jimduff-usds** [comment]
@pawansapra01 ,

The problem is that we pull the organization from the ACTION table, and the ACTION table only has the receiving_org, not the sending_org.

I think the best solution would be to add two new columns to the ACTION table:
    receiving_org and receiving_org_svc
to exactly match the column names in the REPORT_FILE table.

The ACTION table already has columns for the sender (aka submission):  sending_org, and sending_org_client
These are populated in ActionHistory.kt::trackSenderInfo, whcih is called when we receive a file, in ReportFunction.kt::processRequest.

Then we'd need to write a new method ActionHistory.kt::trackReceiverInfo, which is called when we Send a file, in SendFunction.kt.
Make sure this gets saved even in all cases of ACTION.action_name == send, send_warning, and send_error.   This would be a very useful improvement, as its very hard to parse out those failures right now.

Then you'll need to change the code in ReportFileFunction line 168 to figure out whether to use action.sendingOrg or action.receivingOrg.
